catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/UdpStreamProcessor.java

```diff
@@ -325,22 +335,53 @@ public void shutdown() {
         }
     }
 
+    public ExecutorService getRolloverThreadPool() {
+        return rolloverThreadPool;
+    }
+
+    public void setRolloverThreadPool(ExecutorService executorService) {
+        this.rolloverThreadPool = executorService;
+    }
+
     public void checkForRollover() {
         packetBuffer.rotate(rolloverCondition)
                 .ifPresent(this::doRollover);
     }
 
-    public void doRollover(File tempFile) {
-        LOGGER.debug("performing video chunk rollover: tempFile={}", tempFile);
-        try {
-            rolloverAction.doAction(tempFile);
-        } catch (RolloverActionException e) {
-            LOGGER.debug("unable handle rollover file: tempFile={}", tempFile, e);
-        } finally {
-            if (!tempFile.delete()) {
-                LOGGER.debug("unable to delete temp file: filename={}", tempFile);
+    public Future<Void> doRollover(File tempFile) {
+
+        rolloverTaskCount.incrementAndGet();
+        LOGGER.debug(
+                "submitting rollover task to thread pool: submitted task count={} running task count={}",
+                rolloverTaskCount.get(),
+                rolloverThreadCount.get());
+        return rolloverThreadPool.submit(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                rolloverThreadCount.incrementAndGet();
+                LOGGER.debug(
+                        "performing video chunk rollover: tempFile={} submitted task count={} running task count={}",
+                        tempFile,
+                        rolloverTaskCount.get(),
+                        rolloverThreadCount.get());
+                try {
+                    rolloverAction.doAction(tempFile);
```

Comment:
Is `rolloverAction` threadsafe? I see no indication on its interface that it is by contract.